torch/csrc/autograd/variable.h

```diff
@@ -152,8 +164,13 @@ struct TORCH_API Variable : public at::Tensor {
 
   // NOTE: Assignment operators to Tensor come for free from the constructors.
 
-  const at::Tensor& data() const noexcept;
-  at::Tensor& data() noexcept;
+  // NOTE: `var.data()` in C++ has the same semantics as `tensor.data` in Python,
+  // which create a new `Variable` that shares the same storage and tensor metadata
+  // with the original `Variable`, but with a completely new autograd history.
```

Comment:
are you actually using this in this PR?  It might make sense to call this something else temporarily...